pkg/controller/cronjob/utils_test.go

```diff
@@ -379,6 +380,95 @@ func TestGetRecentUnmetScheduleTimes(t *testing.T) {
 	}
 }
 
+func testParseSchedule(t *testing.T, schedule string) cron.SpecSchedule {
+	parsed, err := parseSchedule(schedule)
+	if err != nil {
+		t.Errorf("error parsing schedule %s in test setup: %v", schedule, err)
+	}
+	return parsed
+}
+
+func TestPreviousScheduleTime(t *testing.T) {
+	february29 := func() time.Time {
+		T1, err := time.Parse(time.RFC3339, "2012-02-29T23:30:00Z")
+		if err != nil {
+			panic(err)
+		}
+		return T1
+	}
+
+	april25 := func() time.Time {
+		T1, err := time.Parse(time.RFC3339, "2016-04-25T5:00:00Z")
+		if err != nil {
+			panic(err)
+		}
+		return T1
+	}
+
+	december5 := func() time.Time {
+		T1, err := time.Parse(time.RFC3339, "2015-12-05T23:59:00Z")
+		if err != nil {
+			panic(err)
+		}
+		return T1
+	}
+
+	cases := []struct{
+		schedule string
+		t time.Time
+		expect time.Time
+	} {
```

Comment:
As per the other comment, can we add some tests for [non-standard cron schedules](https://en.wikipedia.org/wiki/Cron#Nonstandard_predefined_scheduling_definitions)?